packages/contracts/src/utils/assertions.ts

```diff
@@ -51,6 +51,18 @@ export function expectRevertOrAlwaysFailingTransactionAsync<T>(p: Promise<T>): P
     return expectRevertOrOtherErrorAsync(p, 'always failing transaction');
 }
 
+/**
+ * Rejects if the given Promise does not reject with the given revert reason or "always
+ * failing transaction" error.
+ * @param p the Promise which is expected to reject
+ * @param reason a specific revert reason
+ * @returns a new Promise which will reject if the conditions are not met and
+ * otherwise resolve with no value.
+ */
+export function expectRevertReasonOrAlwaysFailingTransactionAsync<T>(p: Promise<T>, reason: string): PromiseLike<void> {
```

Comment:
Could we change this to `reason: RevertReasons` to encourage people to add any new reasons to the enum instead of passing a raw string?

Maybe we should also change the name of the enum to the singular `RevertReason` too since a plural name for a type reads a little awkwardly. [Singular names for enum types](https://www.typescriptlang.org/docs/handbook/enums.html) seem more common.